controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -266,6 +271,54 @@ func (r *K8ssandraClusterReconciler) reconcile(ctx context.Context, kc *api.K8ss
 			return ctrl.Result{}, err
 		}
 
+		// reconcile seeds
+		//
+		// The following if block was basically taken straight out of cass-operator. See
+		// https://github.com/k8ssandra/k8ssandra-operator/issues/210 for a detailed
+		// explanation of why this is being done.
+		//
+		// TODO Refactor the if block into a separate method/function.
+		// Note: Some other refactoring needs to be done first so that this can be done cleanly.
+		if len(seeds) > 0 {
+			desiredEndpoints := newEndpoints(desiredDc, seeds)
+			actualEndpoints := &corev1.Endpoints{}
+			endpointsKey := client.ObjectKey{Namespace: desiredEndpoints.Namespace, Name: desiredEndpoints.Name}
+
+			if err = remoteClient.Get(ctx, endpointsKey, actualEndpoints); err == nil {
+				if actualHash, found := actualDc.Annotations[api.ResourceHashAnnotation]; !(found && actualHash == desiredDcHash) {
```

Comment:
1. Shouldn't we be comparing the annotations on `actualEndpoints`?
2. Can be simplified to:

```
if !utils.CompareAnnotations(actualDc, desiredDc, api.ResourceHashAnnotation) {
```